keymanagementrequirements.md

```diff
@@ -0,0 +1,79 @@
+- Repository Owner: In some scenarios a container may be stored in a repository that is managed by the repository owner. The repository may be public or private and could also be air-gapped.
+
+## Definitions:
+- Root key: A self signed key used for the lowest designation of trust. Root keys can be created by developers, organizations, public/private CAs, and registry operators. The root key should be retrieved from a trusted source that can establish the authenticity of the creator's identity.
+- Signing key: A signing key is used to generate artifact signatures. A signing key should be signed with a root key or one of its intermediaries. The certificate chain with a signing key can be used to verify which root it belongs to. While a root key can be used as a signing key, this is not recommended as it creates a large blast radius and increases the risk of compromising a root key. 
+- Trust Policy: The trust policy defines whether to enable signature validation and which checks need to be run. An example trust policy:
+```
+    {
+        "trustPolicy": {
+            "signatureCheck": true,
+            "optionalChecks": {
+            	"signatureExpiry": true,
+            	"signatureRescinded": false
+            },
+            "trustStore": [
+            	"truststore.json",
+            	{
+            		"trustedRoots": [{
+            			"root": "test.crt"
+            		}]
+            	}
+            ],
+            "trustedArtifacts": []
+        }
+    }
+```
+- Trust Store: The trust store defines the relationship between signing keys and artifacts that are used at validation time to determine whether to trust an artifact with a crytptographically valid signature. The trust store will relate a scope (any source, specific registry, specific repository, or specific target) with a certificate (for root key, intermediate, or signing key) or key repository (for automated key distribtuion). It is not recommended to use a signing key as this will cause signature validation to fail if the signing key is rotated. An example trustStore where the first root is trusted for artifacts from any registry and the second root is only trusted for artifacts from "registry.wabbit-networks.io" :
+```
+    {
+    	"trustedRoots": [{
+ 	    		"root": "-----BEGIN CERTIFICATE-----examplecertificate-----END CERTIFICATE-----"
+ 	    	},
+ 	    	{
+ 	    		"root": "-----BEGIN CERTIFICATE-----examplecertificate-----END CERTIFICATE-----",
+ 	    		"scope": "registry.wabbit-networks.io"
+ 	    	}
+ 	    ]
+    }
+```
+- Trust Store Key Information: The key information configured in the trust store will be cryptographically verifiable and contain:
+    - Public Key: Will be used to verify signed artifacts.
+    - Signed Identity: Identity of creator of the signing key.
+    - (Optional) Issuer: If the key is an intermediate this will describe the owner of the root.
```

Comment:
I'd like to see more definition on what it means to "describe the owner of the root". I'm looking for something that can't be easily broken by a bad key (e.g. a lone developer makes their key claim to be the root for all of Docker Hub to make their local mirror work, and doesn't realize how a user configurable field in their key breaks other notary clients). Will this reference point to another key in the trust store? Would it make sense to simply include the full certificate chain any time a key other than a root is added to the store?